### PR TITLE
feat(no_uid_tags): No UID tags anymore 

### PR DESCRIPTION
**Pull request by jeromecambon**
- Removed UID tags
- Enhanced / cleaner CLI handling
- Support of properties file generation from WC name
- Builder pattern for Property and PropertiesInfo

### CODE REVIEW

**Review by benjaminParisel — APPROVED** [review]
Just one small comments, if you don't want to do it, feel free to merge it.
I'm just surprised because you choose to "redevelop" a command line tool instead of re-use some tool already exist

File: README.md [hunk 1]
```diff
@@ -2,151 +2,197 @@
...
+cwb gen-properties -webComponent <web component source file> [-outputDir <directory>]
+```
+or
+```shell
+cwb gen-properties -webComponentName <web component name> [-outputDir <directory>]
 ```
 
+#### Generate Custom Widget:
+```shell
+cwb gen-widget -properties <json properties file> [-outputDir <directory>]
```
[the remark this answers is not in the document]
Comment (jeromecambon) [reply README.md:184]: agree

**Comment by jeromecambon** [comment]
> Just one small comments, if you don't want to do it, feel free to merge it.
> I'm just surprised because you choose to "redevelop" a command line tool instead of re-use some tool already exist

It seems simple enough to avoid to add a dependency. But perhaps if the CLI become more complex... Do you have in mind a simple lib ?

File: custom-widget-builder/src/CliHandler.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  private checkCommand() {
+    this.command = this.params[0];
+    if (!this.command ||
+      (this.command !== CliHandler.genPropertiesCommand && this.command !== CliHandler.genWidgetCommand)) {
+      this.usage();
+    }
+  }
+
+  private checkParamsMap() {
+    if (this.paramsMap.size > 2) {
```
[the remark this answers is not in the document]
Comment (jeromecambon) [reply custom-widget-builder/src/CliHandler.ts:101]: ok

File: README.md [hunk 3]
```diff
@@ -2,151 +2,197 @@
...
-   * -@bond constant
-   * @type {choice}
-  */
-  @property({ attribute: 'type', type: String, reflect: true })
-  type: string = "text";
+### Using the CLI
 
+#### Generate properties file:
+```shell
+cwb gen-properties -webComponent <web component source file> [-outputDir <directory>]
```
Comment (benjaminParisel) [inline README.md:175]:
```suggestion
cwb gen-properties -webComponentFile <web component source file> [-outputDir <directory>]
```

File: README.md (hunk 1 above)
Comment (benjaminParisel) [inline README.md:184]:
```suggestion
cwb gen-widget -propertiesFile <json properties file> [-outputDir <directory>]
```

**Pull request merged**